lib/withOnyx.js

```diff
@@ -28,37 +29,51 @@ export default function (mapOnyxToState) {
     return (WrappedComponent) => {
         const displayName = getDisplayName(WrappedComponent);
         class withOnyx extends React.Component {
+            pendingSetStates = [];
+
             constructor(props) {
                 super(props);
-
+                this.delayUpdates = !!options.delayUpdates;
                 this.setWithOnyxState = this.setWithOnyxState.bind(this);
+                this.flushPendingSetStates = this.flushPendingSetStates.bind(this);
 
                 // This stores all the Onyx connection IDs to be used when the component unmounts so everything can be
                 // disconnected. It is a key value store with the format {[mapping.key]: connectionID}.
                 this.activeConnectionIDs = {};
 
-                const cachedState = _.reduce(mapOnyxToState, (resultObj, mapping, propertyName) => {
-                    const key = Str.result(mapping.key, props);
-                    const value = Onyx.tryGetCachedValue(key, mapping);
-
-                    /**
-                     * If we have a pending merge for a key it could mean that data is being set via Onyx.merge() and someone expects a component to have this data immediately.
-                     *
-                     * @example
-                     *
-                     * Onyx.merge('report_123', value);
-                     * Navigation.navigate(route); // Where "route" expects the "value" to be available immediately once rendered.
-                     *
-                     * In reality, Onyx.merge() will only update the subscriber after all merges have been batched and the previous value is retrieved via a get() (returns a promise).
-                     * So, we won't use the cache optimization here as it will lead us to arbitrarily defer various actions in the application code.
-                     */
-                    if (value !== undefined && !Onyx.hasPendingMergeForKey(key)) {
-                        // eslint-disable-next-line no-param-reassign
-                        resultObj[propertyName] = value;
-                    }
+                const cachedState = _.reduce(
+                    mapOnyxToState,
+                    (resultObj, mapping, propertyName) => {
+                        const key = Str.result(mapping.key, props);
+                        let value = Onyx.tryGetCachedValue(key, mapping);
+                        if (!value && mapping.initialValue) {
```

Comment:
reminder for @ospfranco (because he is new to the repo) that these docs are generated:

https://github.com/Expensify/react-native-onyx/blob/bf3ffa5f934e0197590997f945f40ab6ddf704e1/package.json#L40

Reply:
Will do! thanks!